Common/include/drivers/CDriverBase.hpp

```diff
@@ -0,0 +1,640 @@
+/*!
+ * \file CDriverBase.hpp
+ * \brief Base class template for all drivers.
+ * \author H. Patel, A. Gastaldi
+ * \version 7.4.0 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2022, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "../../../SU2_CFD/include/numerics/CNumerics.hpp"
+#include "../../../SU2_CFD/include/output/COutput.hpp"
+#include "../../../SU2_CFD/include/solvers/CSolver.hpp"
+#include "../CConfig.hpp"
+
+class CDriverBase {
+ protected:
```

Comment:
This should be in SU2_CFD instead of Common, and it is missing doxygen docs

Reply:
`CDriverBase` is in Common because it is not exclusively used for SU2_CFD drivers. The SU2_DEF module has `CDeformationDriver` and `CDiscAdjDeformationDriver` that also inherit from `CDriverBase`. In the future, other SU2 modules can also build off of `CDriverBase`

Reply:
Noted about doxygen docs!

Reply:
But it is including things from SU2_CFD... which defeats the purpose of having Common.
Move this to SU2_CFD and make the deformation drivers depend on SU2_CFD.

Reply:
CDriverBase was moved to SU2_CFD directory in commit [5c1ec48](https://github.com/su2code/SU2/pull/1300/commits/5c1ec48ef8f712b67afe88faa8011ace8282cfc6)